### ISSUE TITLE
Re-enable `prefer-json-parse-buffer`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2565]
### Description

This rule actually makes sense to JavaScript users, but it's disabled due to compatibility with TypeScript, how about rename to `consistent-json-file-read`, and default to read as `string`, but add option to allow read it as `buffer`?

### Examples

```js
// ❌
JOSN.parse(await fs.readFile('/path/to/file.json'));

// ✅
JOSN.parse(await fs.readFile('/path/to/file.json', 'utf8'));
```

```js
// ❌
// eslint unicorn/consistent-json-file-read: ["error", "buffer"]
JOSN.parse(await fs.readFile('/path/to/file.json', 'utf8'));

// ✅
// eslint unicorn/consistent-json-file-read: ["error", "buffer"]
JOSN.parse(await fs.readFile('/path/to/file.json'));
```

### Additional Info

We may want check other cases in the future, since there are more way to read a JSON file now.

```js
createRequire(import.meta.url)('/path/to/file.json');
```

```js
await import('./path/to/file.json', {with:{type:'json'}});
```
